src/sync/pull-request.ts

```diff
@@ -53,10 +57,14 @@ export const getPullRequestTask = async (
 	perPage: number,
 	messagePayload: BackfillMessagePayload
 ) => {
+	if (await booleanFlag(BooleanFlags.USE_NEW_PULL_ALGO, jiraHost)) {
+		return doGetPullRequestTask(logger, gitHubInstallationClient, jiraHost, repository, messagePayload, cursor as string, perPage);
+	}
+
 	const smartCursor = new PageSizeAwareCounterCursor(cursor).scale(perPage);
 	const numberOfPagesToFetchInParallel = await numberFlag(NumberFlags.NUMBER_OF_PR_PAGES_TO_FETCH_IN_PARALLEL, 0, jiraHost);
```

Comment:
(not in this PR) Wondering if we still need parallel fetching... 🤔

Reply:
Or will it be removed after this PR is merged and FF is cleaned up?

Reply:
not for the gql, but its doing so much less work it may not need to, I didn't want to overcomplicate the first change